### ISSUE TITLE
Completed 100 Days of Code 🎉   (08/11)

### ISSUE THREAD

**Issue by anitabe404** [studydash/cards#300]
## Motivation/Problem Statement 💭❓:
Today, I am here to put in my 100th consecutive code session.

---

## Today's Goal/Solution 🥅:
I'll be working on Kaggle's 30 Days of Machine Learning challenge.

---

## Result 🍓🍌🍉:
I completed Day 10 of 30 Days of ML. Time is flying because that means I'm 33% complete with the Kaggle challenge. I feel like I just signed up. Today I learned about a new model: Random Forest. I am definitely in the deep end right now. I'm at the point where I'm using the models, but I don't fully get them yet. It's still cool though.

---

## Observations & Next Steps 🔭👣:
I will continue on with 30 Days of ML. I'm also exploring some other sites that have data science tutorials and instructions. I'm still undecided on my next move.

---

## References 🔗:
Link any useful resources you may have found throughout the day.

---

## Extra/Fun (Optional) 🎈🎉🥳:
I spent a good hour and a half outside soaking up the sun. I also had a good chat with a tech twitter friend. I'm getting to know more and more people. Overall, it was a really good day. 😌

**Comment by r002** [comment]
(Omg, this took me a positively _**absurd**_ amount of time to glue together in Photoshop and PowerPoint... 😅) But lo and behold...

# _Congratulations, Anita!!_

## You are **_the first member ever_** to complete `100 Days of Code` in `StudyDash`!!  Fantastic job and congratulations!!  _Well done!_

# You're a legend!! 🎉🎈🥳

@anitabe404 Thank you **_so_** much for doing this past `100 days of code` with me on [`StudyDash`](https://studydash.github.io/)!  Ever since we met on [CodeNewbie](https://community.codenewbie.org/r002/5-codenewbie-study-group-cohort-looking-for-study-mates-4lpj) back on that fateful day at the end of spring, you've diligently stuck the entire duration!  Others have come and gone, but you endured and now, at long last, we have crossed the finish line!  Truly, I enjoy just browsing our wall of `cards` and seeing everything we've learned and accomplished together since the [beginning of May](https://studydash.github.io/v1/?d=2021-05).

- [Your first Google mockup sketch](https://github.com/studydash/cards/issues/7#issuecomment-832916463)! ✒
- That one time [you coded with no power!](https://github.com/studydash/cards/issues/111) 😳
- [Your first `C#` program!](https://github.com/studydash/cards/issues/202) 🔥
- That gorgeous [`Python CLI app`](https://github.com/studydash/cards/issues/222#issue-948901437)! 😀
- The [first time](https://github.com/studydash/cards/issues/296) you built a decision tree and used pandas & scikit-learn for Kaggle! 🍾

Anita, it's honestly been an honor and privilege to have met and have spent this summer in your company.  We've never met IRL but I feel like we've been in the trenches together these past hundred days and think of you as a fellow comrade in arms. 💂✊  Always feel free to ping me anytime about anything and definitely keep in touch-- I look forward to seeing what you do next!  You are destined for greatness! 🙂🚀✨

![image](https://user-images.githubusercontent.com/45280066/129138610-e533a477-8b31-425f-a37a-bed4452c9f9e.png)

[Download ➡ PDF!](https://studydash.github.io/certificates/Anita%20Beauchamp%20-%20StudyDash%20-%20CERTIFICATE%20OF%20COMPLETION%20(Aug%2011,%202021).pdf) 🏆

**Comment by anitabe404** [comment]
@r002 OMG!!!!!! This is so nice.

You have no idea how much it's meant to me to be part of this study group. There were a few days where I was starting to feel down and you always pushed me to keep going.

I wasn't expecting this certificate and honestly, it's the best thing ever. I really, really appreciate it.

I can't believe that it's been 100 days. I remember being so nervous about being able to keep up. I'm so happy you replied to my post with the offer to be part of study dash.

I definitely want to continue being a part of the group as long as it's going. I will be taking a break though before I hop back in the saddle. Although, maybe not as long as I was planning since I'm in the middle of this Kaggle challenge. 🥴

**Comment by shazahuang** [comment]
Congratulations!! We are so proud of you!  😊

**Comment by r002** [comment]
@anitabe404 Of course!  It was wonderful having you onboard!  And you seriously helped me a ton too!  We pushed each other, as all good study groups do. 🤗  You'll always have a column here on `StudyDash` as long as you want it!  🙂  

I don't know of any specific next steps yet, but I do plan, for now at least, to keep personally contributing cards daily.  I really enjoy using these GitHub `cards` as a form of daily journaling.  I just love seeing that `Wall of Progress` and also seeing the cards _**link**_ to each other when they explicitly reference one another.  Over time, it becomes a neat `knowledge graph` of sorts.  Very gratifying!  😁  And also seeing the journeys of others too as well!  🎉

This weekend though, I'm gonna try to implement a new feature where study members can set themselves as `Active` or `Inactive` (or something like that).  While I was planning on implementing `OAuth` at some point (ie. you'd have to actually login with your GitHub login to set personalized options), I'm not sure if I want to do that _yet_.  So short of that, I need to see if I can think up another way of implementing personalization without OAuth. 🤔  This entire project has been enormously organic (we're just making it up as we go! 😅) but, in general:  It makes sense to me that sometimes people will be in `Campaign mode` and on a mission to complete `X consecutive days`.  But eventually, campaigns **_finish_**, people need to take breaks, check out, etc.  And then have the ability to resume _later_ when they want to embark upon _another_ campaign with a different cohort.

And so the card statuses and colors need to reflect this.  Like if one's on `Inactive Mode`, then thou shouldn't be racking up `Missed Days` and having it mess up one's `streaks`.  Basically:  The `stats` and `metrics` (and entire GUI) need to reflect the `Active`/`Inactive` status of a study member.

Aghast, so much to do and so little time!  😣  In any case, feel free to continue contributing cards any time you wish!  `StudyDash` and I will be here! 😊

**Comment by anitabe404** [comment]
@shazahuang Thank you!!! It's been such a good time in study dash.

**Comment by anitabe404** [comment]
@r002 You're always building the best new features for us. I think your plan to add an inactive mode would really come in handy. I'm thinking of trying a hand at a python script that will create my cards for me, lol. Sometimes I just space and I miss adding them before midnight when the carriage turns into a pumpkin.

**Comment by r002** [comment]
@anitabe404 Very neat! 😄 Yes, scripting/automation is definitely one of the sweet spots of coding.  When we use code to make our lives easier, it definitely feels super-satisfying!  🎉  I know you're in the middle of Kaggle right now, but if you do tackle writing a Python script later and need any help, lmk!  I've already [written code](https://github.com/r002/storyline-api/blob/58e65fcf2b659f682af99a510512a1a75cd51477/ghservices/dataservice.go#L60-L103) (albeit, in `Go`; and omg, looking at that now; the code's a bit messy 😅🤦‍♂️) that updates cards programmatically via the `GitHub REST API`.  For example, you may have noticed that every time in `StudyDash` you submit a card, the `studygroup-bot` adds the `Daily Accomplishment` milestone and also labels the card with the `weekday`.  All of that is done programmatically by the server.  Eg.

![image](https://user-images.githubusercontent.com/45280066/129387745-bacdcf50-684c-43e0-b772-991e4c5c975e.png)

The crux of it is that you [model](https://github.com/r002/storyline-api/blob/58e65fcf2b659f682af99a510512a1a75cd51477/ghservices/models.go#L33-L44) (usually a subset of) the `payload` you wish to send (in this case, a [`GitHub Issue`](https://docs.github.com/en/rest/reference/issues#create-an-issue)) and then you can just programmatically fire off `http requests` to do your bidding! 😁.  There's also some `authentication` stuff but GitHub's written some pretty good [docs](https://docs.github.com/en/rest/guides/basics-of-authentication) and guides for that.

Anyway, feel free to ping me if you ever need any help!  You could totally add a `Python StudyDash Card Submitter` feature to your [`XDaysOfYTracker`](https://github.com/anitabe404/XDaysOfYTracker) app! 🥳
